src/components/misc/personViews/PersonViewTable.jsx

```diff
@@ -132,7 +132,24 @@ export default class PersonViewTable extends React.Component {
                             const val0 = row0.data.content[this.state.sortIndex] || '';
                             const val1 = row1.data.content[this.state.sortIndex] || '';
 
-                            let x = val0.toString().localeCompare(val1.toString());
+                            let x = 0;
+
+                            if (Array.isArray(val0) && typeof(val0[0]) != 'undefined' && typeof(val1[0]) != 'undefined') {
+                                x = val0[0].text.toLowerCase().toString().localeCompare(val1[0].text.toLowerCase().toString());
```

Comment:
On this line you're doing `toLowerCase().toString()` before `localeCompare()`. I have a few comments on this.

## `toString()` is unnecessary
First, the `toLowerCase()` method exists on `String` and returns a `String`, so if that method is available you are already working with a string and there is no need to do `toString()` explicitly. This would leave you with the following instead:

```js
// Shorter without explicit tostring()
x = val0[0].text.toLowerCase().localeCompare(val1[0].text.toLowerCase());
```

Next it turns out that `toLowerCase()` is not necessary, and in fact [not desirable](https://stackoverflow.com/a/49004987/229824) in this scenario. Let's investigate why!

## Do we need `toLowerCase()`?
Without `localeCompare()` (which hasn't always existed in javascript), sorting an array of strings will yield a result where all uppercase strings come before the lowercase ones, like so:

```js
['abc', 'DEF', 'def'].sort();
// Result: ['DEF', 'abc', 'def']
````

It does this because the uppercase characters 'ABCDEFG…' come before the lowercase equivalents 'abcdefg…' in [the ASCII table](http://ascii-table.com/), which is the original lookup table used by computers to convert between binary data and human-readable characters (nowadays it's been extended by Unicode and other codecs to include more characters, symbols, emojis etc).

The default sorting behaviour is obviously undesirable, because any human would expect 'abc' to come before 'DEF'. A solution to this has been to use `toLowerCase()` so that all strings were compared as lower-case:

```js
['DEF', 'abc', 'def'].sort((s0, s1) => {
  if (s0.toLowerCase() > s1.toLowerCase()) return 1;
  if (s1.toLowerCase() > s0.toLowerCase()) return -1;
  else return 0;
})
// Result: ['abc', 'DEF', 'def']
```

This is better, but it still doesn't solve a very big problem, which is that we need to work with strings that are not restricted to the 26-character basic English alphabet, i.e. the characters listed in the ASCII table. Where do we sort a string that begins with 'å'? Towards the end, right? It's obvious for Swedes, but not for computers. By default javascript (and most environments) will treat it as an 'a', and not put it at the end of a sort as we wold expect.

Enter `localeCompare()`. It's main purpose is to handle localization, i.e. different languages. But it also has the added benefit that it will treat 'd' and 'D' as related characters. This means that sorting will be much more human-friendly:

```js
['def', 'abc', 'DEF'].sort((s0, s1) => s0.localeCompare(s1));
// Result: ['abc', 'def', 'DEF']
```

This is exactly what we'd like! So it turns out we don't need `toLowerCase()` at all.

## Is `localeCompare()` case insensitive?
Does this mean that `localeCompare()` ignores lowercase vs uppercase? No, it does not. It will always prioritize lower-case characters over upper-case ones. Sorting with `localeCompare()` will put a string beginning with 'd' before the same string beginning with 'D'. But it will never (in English or Swedish at least) put 'd' or 'D' before 'c' or 'C'.

## Ditching `toLowerCase()`
Knowing this (and provided that we don't care about true case-insensitive sorting, which I don't think we have to) we can rewrite this line of code to look like this:

```js
x = val0[0].text.localeCompare(val1[0].text);
```

This is of course shorter, but I think also easier to read.